### ISSUE TITLE
Regression of getting iterator after a deletion by range

### ISSUE THREAD

**Issue by aleksuss** [facebook/rocksdb#10662]
Hello. I encountered a strange behaviour (imho) of getting an iterator after deleting keys from an empty database by range. The time of getting an iterator increased with time. But calling `db->Flush(rocksdb::FlushOptions());` before getting an iterator seems fixes the issue. Is this an expected behaviour and what's the right way to handle such behaviour? Thanks.

### Expected behaviour
The time of getting an iterator **isn't** increased.

### Actual behaviour
The time of getting an iterator **is** increased.

### Steps to reproduce the behaviour
Example of code to reproduce the issue:
```cpp
#include <chrono>
#include <thread>
#include <rocksdb/db.h>

int main() {
    rocksdb::Options options;
    options.create_if_missing = true;
    rocksdb::DB *db;
    rocksdb::DB::Open(options, "/tmp/testdb", &db);

    auto cf = db->DefaultColumnFamily();
    rocksdb::Slice from(std::to_string(42));
    rocksdb::Slice to(std::to_string(44));


    while (true) {
        auto wb = new rocksdb::WriteBatch();
        wb->DeleteRange(cf, from, to);
        db->Write(rocksdb::WriteOptions(), wb);

        std::this_thread::sleep_for(std::chrono::milliseconds(50));
        // db->Flush(rocksdb::FlushOptions()); // Fixes the issue !!!!!!!
        auto start = std::chrono::high_resolution_clock::now();
        auto _iter = db->NewIterator(rocksdb::ReadOptions());
        auto stop = std::chrono::high_resolution_clock::now();
        std::chrono::duration<float> duration = stop - start;

        printf("time: %.6f secs\n", duration.count());
    }
}
```

**Comment by cbi42** [comment]
Hi, this is expected, some [comment](https://github.com/facebook/rocksdb/blob/be09943fb58a2dd3f70e6e30781ebfa3fcbcb8fa/include/rocksdb/db.h#L472) on this from API documentation, and the suggestion is to do manual flush occasionally as you found out already. Internally, range tombstones from a live memtable are fragmented each time a new iterator is created (code [here](https://github.com/facebook/rocksdb/blob/be09943fb58a2dd3f70e6e30781ebfa3fcbcb8fa/db/memtable.cc#L565)). This can cause iterator creation regression as the number of range tombstone accumulates. There is work on improving this (#10547). If that is merged, then the regression will only happen at the first iterator creation (after a new range deletion), the following ones will be fast.

**Comment by aleksuss** [comment]
Thanks for your response. But the flush on every iteration creates a new sst file.

**Comment by aleksuss** [comment]
Hello @cbi42. Has the issue been fixed in the 7.6.0 version?

**Comment by ajkr** [comment]
#10547 will be in the 7.7 minor release series, for which there are currently no tags. Based on typical past timelines I'd expect a tagged 7.7 release in ~3 weeks.
